sdk/storage/storage-queue/review/storage-queue.api.md

```diff
@@ -252,14 +258,13 @@ export type MessagesClearResponse = MessagesClearHeaders & {
 export interface MessagesDequeueHeaders {
     clientRequestId?: string;
     date?: Date;
-    // (undocumented)
     errorCode?: string;
     requestId?: string;
     version?: string;
 }
 
 // @public
-export interface MessagesDequeueOptionalParams extends coreHttp.RequestOptionsBase {
+export interface MessagesDequeueOptionalParams extends coreHttp.OperationOptions {
```

Comment:
This is a breaking change that I'm not sure we should take. Many options move, some options are no longer supported. I dislike ROB as much as anyone, but this change worries me. Thoughts?

Reply:
@joheredi Looping you for your thoughts on this.

Reply:
yes, at least `spanOptions` is wrapped in `tracingOptions` in `OperationOptions`

Reply:
Yeah, this is probably a big breaking change that we should try to avoid. I think we'll have to handle the conversion from ROB to OperationOptions at the convenience layer, so public methods keep taking the same shape that extends RequestOptionsBase and in our convenience layer build the object that extends OperationOptions before feeding it to the generated client.

`core-http` has a method `operationOptionsToRequestOptionsBase` we may need something similar but in the opposite direction.

Reply:
@bterlson @jeremymeng @joheredi Added requestOptionsBaseToOperationOptions method with the latest commit. Now, this has changed back to the previous format with no breaking change